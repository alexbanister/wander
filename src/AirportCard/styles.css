.airportListContainer {
  position: relative;
  width: 100%; }

.airportList {
  position: absolute;
  width: 100%; }

.singleAirport {
  font-size: 1rem;
  background: #FFFFFF;
  color: #384E5E;
  border: 1px solid #384E5E;
  cursor: pointer;
  padding: .75em;
  display: grid;
  grid-template-columns: 3fr 6fr 1fr;
  grid-template-areas: "iata name remove" "iata city remove"; }
  .singleAirport:hover {
    background: #F6D22F; }
  .singleAirport .IATAcode {
    grid-area: iata;
    font-size: 2em; }
  .singleAirport .airportName {
    grid-area: name; }
  .singleAirport .airportCity {
    grid-area: city; }
