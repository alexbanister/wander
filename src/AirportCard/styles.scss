@import "../variables.scss";

.airportListContainer {
  position: relative;
  width: 100%;
}
.airportList {
  position: absolute;
  width: 100%;
}
.singleAirport {
  font-size: 1rem;
  background: $background;
  color: $body-dark;
  border: 1px solid $body-dark;
  cursor: pointer;
  padding: .75em;
  display: grid;
  grid-template-columns: 3fr 6fr 1fr;
  grid-template-areas: "iata name remove" "iata city remove";
  &:hover {
    background: $highlight;
  }
  .IATAcode {
    grid-area: iata;
    font-size: 2em;
  }
  .airportName {
    grid-area: name;
  }
  .airportCity {
    grid-area: city;
  }
}
