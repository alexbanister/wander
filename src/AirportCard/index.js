import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const AirportCard = ({ airport, onClick, removeFlag }) => {
  let removeButton = removeFlag ?
    <span className='destinationRemove' onClick={onClick}>X</span> :
    '';
  let addClick = removeFlag ? ()=>{} : onClick;
  return (
    <div className='singleAirport' onClick={addClick}
      key={airport.iata}>
      <div className='IATAcode'>{airport.iata}</div>
      <div className='airportName'>{airport.name}</div>
      <div className='airportCity'>
        {airport.city}, {airport.state.abbr}, {airport.country.name}
      </div>
      {removeButton}
    </div>
  );
};

AirportCard.propTypes ={
  onClick: PropTypes.func,
  removeFlag: PropTypes.bool,
  airport: PropTypes.object
};

export default AirportCard;
