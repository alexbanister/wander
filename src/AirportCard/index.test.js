import { shallow } from 'enzyme';
import AirportCard from './index';
import React from 'react';


describe('AirportCard', () => {
  it('should always match the snapshot', () => {
    const airport = {
      iata: 'DEN',
      name: 'Denver',
      city: 'Denver',
      state: {
        abbr: 'CO'
      },
      country: {
        name: 'USA'
      }
    };
    const wrapper = shallow(<AirportCard
      airport={airport}
      onClick={jest.fn()}
      removeFlag={false}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
