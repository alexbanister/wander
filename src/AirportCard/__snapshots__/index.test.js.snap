// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirportCard should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AirportCard
    airport={
        Object {
            "city": "Denver",
            "country": Object {
              "name": "USA",
            },
            "iata": "DEN",
            "name": "Denver",
            "state": Object {
              "abbr": "CO",
            },
          }
    }
    onClick={[Function]}
    removeFlag={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "DEN",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="IATAcode"
>
          DEN
</div>,
        <div
          className="airportName"
>
          Denver
</div>,
        <div
          className="airportCity"
>
          Denver
          , 
          CO
          , 
          USA
</div>,
        "",
      ],
      "className": "singleAirport",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "DEN",
          "className": "IATAcode",
        },
        "ref": null,
        "rendered": "DEN",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Denver",
          "className": "airportName",
        },
        "ref": null,
        "rendered": "Denver",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Denver",
            ", ",
            "CO",
            ", ",
            "USA",
          ],
          "className": "airportCity",
        },
        "ref": null,
        "rendered": Array [
          "Denver",
          ", ",
          "CO",
          ", ",
          "USA",
        ],
        "type": "div",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "DEN",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="IATAcode"
>
            DEN
</div>,
          <div
            className="airportName"
>
            Denver
</div>,
          <div
            className="airportCity"
>
            Denver
            , 
            CO
            , 
            USA
</div>,
          "",
        ],
        "className": "singleAirport",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "DEN",
            "className": "IATAcode",
          },
          "ref": null,
          "rendered": "DEN",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Denver",
            "className": "airportName",
          },
          "ref": null,
          "rendered": "Denver",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Denver",
              ", ",
              "CO",
              ", ",
              "USA",
            ],
            "className": "airportCity",
          },
          "ref": null,
          "rendered": Array [
            "Denver",
            ", ",
            "CO",
            ", ",
            "USA",
          ],
          "type": "div",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
